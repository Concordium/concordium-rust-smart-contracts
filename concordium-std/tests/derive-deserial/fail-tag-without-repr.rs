//! Ensure `derive(Deserial)` fails when using 'tag' attribute without a
//! 'repr' attribute.
use concordium_std::*;

#[derive(Deserial)]
enum MyTaggedEnum {
    #[concordium(tag = 42)]
    One,
    Two,
}

fn main() {}
