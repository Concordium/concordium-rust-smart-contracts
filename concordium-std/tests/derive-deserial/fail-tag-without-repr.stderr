error: 'repr(..)' attribute must be set on the type when using 'tag' attribute
 --> tests/derive-deserial/fail-tag-without-repr.rs:5:10
  |
5 | #[derive(Deserial)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `Deserial` (in Nightly builds, run with -Z macro-backtrace for more info)
